src/twisted/internet/unix.py

```diff
@@ -171,10 +171,22 @@ def doRead(self):
             else:
                 return main.CONNECTION_LOST
 
-        if ancillary:
-            ancillaryData = ancillary[0][2]
-            fdCount = len(ancillaryData) // 4
-            fds = struct.unpack('i'*fdCount, ancillaryData)
+        for cmsgLevel, cmsgType, cmsgData in ancillary:
+            if (cmsgLevel != socket.SOL_SOCKET or
+                cmsgType != sendmsg.SCM_RIGHTS):
+                log.msg(
+                    format=(
+                        "%(protocolName)s (on %(hostAddress)r) "
+                        "received unsupported ancillary data "
+                        "(level=%(cmsgLevel)r, type=%(cmsgType)r) "
+                        "from %(peerAddress)r."),
+                    hostAddress=self.getHost(), peerAddress=self.getPeer(),
+                    protocolName=self._getLogPrefix(self.protocol),
+                    cmsgLevel=cmsgLevel, cmsgType=cmsgType,
+                    )
+                continue
+            fdCount = len(cmsgData) // 4
+            fds = struct.unpack('i'*fdCount, cmsgData)
```

Comment:
PS: Had to rename the `_ancillary_SOL_SOCKET_SCM_RIGHTS` method to comply with twistedchecker.